File: org-frontend/src/api/model/note.ts
// eslint-disable-next-line no-restricted-imports
import dayjs, { Dayjs } from "dayjs"
import { Tag } from "./tag"

export interface Note {
  id: string
  userId: string
  title: string
  content: string
  createdAt: Dayjs
  updatedAt: Dayjs
  publishedAt: Dayjs | undefined
  published: boolean
  isEditable?: boolean
  tags: Tag[]
}

export function fromJson(note: Note): Note {
  return {
    ...note,
    createdAt: dayjs(note.createdAt),
    updatedAt: dayjs(note.updatedAt),
    publishedAt: note.publishedAt ? dayjs(note.publishedAt) : undefined,
    tags: note.tags || [],
  }
}

File: sigil-go/pkg/models/note.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Note struct {
	ID          uuid.UUID  `json:"id"          db:"id"`
	UserID      uuid.UUID  `json:"userId"      db:"user_id"`
	Title       string     `json:"title"       db:"title"`
	Content     string     `json:"content"     db:"content"`
	CreatedAt   time.Time  `json:"createdAt"   db:"created_at"`
	UpdatedAt   time.Time  `json:"updatedAt"   db:"updated_at"`
	PublishedAt *time.Time `json:"publishedAt" db:"published_at"`
	Published   bool       `json:"published"   db:"published"`
	Tags        []Tag      `json:"tags" db:"-"`
}
